packages/core/src/components/menu/menuItem.tsx

```diff
@@ -127,10 +127,9 @@ export interface MenuItemProps
     popoverProps?: Partial<Omit<PopoverProps, "content" | "minimal">>;
 
     /**
-     * Whether this item should appear selected.
-     * Defining this  will set the `aria-selected` attribute and apply a
-     * "check" or "blank" icon on the item (unless the `icon` prop is set,
-     * which always takes precedence).
+     * Whether this item should appear selected - `roleStructure` must be `"listoption"` for this to be
```

Comment:
Thank you for adding this comment!

> `roleStructure` must be `"listoption"` for this to be applied.

This prop being a no-op depending on `roleStructure` would have surprised me too, especially since `"listoption"` isn't the default.